src/Lib.hs

```diff
@@ -1,8 +1,283 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE RecordWildCards            #-}
+
 module Lib
-       ( someFunc
-       ) where
+    ( runZendeskMain
+    , collectEmails
+    , processTicket
+    , processTickets
+    , fetchTickets
+    , showStatistics
+    ) where
 
 import           Universum
--- TODO(ks): Here we want to import the functionality we require in cardano-report-server.
-someFunc :: IO ()
-someFunc = putTextLn "someFunc"
+
+import           Data.Text (stripEnd, isInfixOf)
+import           Data.Attoparsec.Text.Lazy (eitherResult, parse)
+
+import           LogAnalysis.Types
+import           LogAnalysis.Classifier
+import           LogAnalysis.KnowledgeCSVParser
+import           Util
+import           CLI
+import           Zendesk
+
+------------------------------------------------------------
+-- Functions
+------------------------------------------------------------
+
+runZendeskMain :: IO ()
+runZendeskMain = do
+    args <- getCliArgs
+    putTextLn "Welcome to Zendesk classifier!"
+    token <- readFile tokenPath  -- Zendesk token
+    assignFile <- readFile assignToPath  -- Select assignee
+    knowledges <- setupKnowledgebaseEnv knowledgebasePath
+    assignTo <- case readEither assignFile of
+        Right agentid -> return agentid
+        Left  err     -> error err
+    let cfg' = defaultConfig
+                   { cfgToken = stripEnd token
+                   , cfgAssignTo = assignTo
+                   , cfgKnowledgebase = knowledges
+                   }
+    let getAgentId = zlGetAgentId . cfgZendeskLayer $ cfg'
+    agentId <- runApp getAgentId cfg'
+    let cfg = cfg' { cfgAgentId = agentId }
+    -- At this point, the configuration is set up and there is no point in using a pure IO.
```

Comment:
I'm not sure what this means (Not enough haskell knowledge to understand it).  Can you explain it to me?

Reply:
It just means that there is no point in using `IO` all over the codebase since the `Config` is initialized and we require `Config` to use the `App` newtype.
The consequences of this are - try to see how many `runApp` were removed...